### ISSUE TITLE
Address root prior support for ambiguous gene dynamics

### ISSUE THREAD

**Issue by VolkerBergen** [theislab/scvelo#368]
The following PR needs to be adapted to the current implementation and extended with different types of priors (root cell or a set of potential root cells such as first expt timepoint). 

https://github.com/theislab/scvelo/pull/144

**Comment by ZheFrench** [comment]
Hello,
Thanks for this soft. There is a lot of work here.
I would like to know if is finally possible to  set the root clusters (or endpoint) for running scVelo ?
I would like to reverse the latent time given by scvelo which is the opposite I expected.
I tried to use root_key from latent.time (passing cells ID with `adata.obs['root_cells']` #114) but without effect on the latent time graphic. Is this the good way to go ? Is there a use case somewhere ? I know it has been asked several times  in the past but didn't find a clear answer and also the version has evolved so ...
Thanks.

**Comment by WeilerP** [comment]
@ZheFrench, this has not yet been developed and I am also not sure I will do so as this can easily lead to false security/conclusions.
Either way, specifying a root cell works. I just double checked and I got it to work  by specifying the numerical ID of the root cell (it's a bit hacky), _i.e._, something along the lines of

```python
df = pd.DataFrame(index=adata.obs_names).reset_index()
adata.uns['root_key'] = df.index[df['index'].isin(adata.obs_names[adata.obs[COLUMN_NAME] == CLUSTER_LABEL])][0]

scv.tl.latent_time(adata, root_key='root_key')
```

**Comment by ZheFrench** [comment]
Did not work for me.
Ok could you give juste the content of  `adata.uns['root_key'] ` I mean the format. By numerical ID , you mean the index not the cellID ?
It's just one cell ?

I update a little bit,seems better but run for ages whith no end.
```
sample_one_index = pd.DataFrame(sample_one.obs.index)
sample_one.uns['root_key'] = sample_one_index[sample_one_index.isin(sample_one.obs_names[sample_one.obs['my_cluster'] == 1])]
print(sample_one.uns['root_key'])
```
  

> CellID
> 0                              NaN
> 1                              NaN
> 2                              NaN
> 3                              NaN
> 4                              NaN
> ...                            ...
> 2991                           NaN
> 2992                           NaN
> 2993  TTTGTTGTCTCTGGTCx
> 
> 

Thanks

**Comment by WeilerP** [comment]
> Did not work for me. 
I cannot help you without some information (logs, expected vs. actual behavior, etc.)

> Ok could you give juste the content of `adata.uns['root_key'] ` I mean the format. By numerical ID , you mean the index not the cellID ? It's just one cell ?

Yes, there is only one root cell, not an entire cluster. If in your AnnData object, the 10th cell is the root cell, you should do

```python
adata.uns['root_key'] = 9
```

This is what I meant by numerical ID (sorry, should have been more clear on this). My code snippet from above takes a given cluster annotation stored in `.obs[COLUMN_NAME]` and selects all cells which fall into the cluster `CLUSTER_LABEL` where your root cell lies (e.g. stem cells). I then simply selected the very first cell of all these candidate cells (as I said before, this is a bit hacky and very rudimentary).

**Comment by ZheFrench** [comment]
Ok it worked. It reversed the color for the latent time plotting. Perfect.
But then there is no way, to reverse the direction of the arrows from the velocity graph ?

**Comment by WeilerP** [comment]
No, there is no way to reverse the arrow direction. #216 or #462 might be useful to understand why the inferred directionality is inconsistent with prior knowledge/what you expect.

**Comment by ZheFrench** [comment]
Thanks, it's a bit far away from the initial question. 
But I just would like to ask : What's the latent time meaning ? I don't understand. Sorry to bother you...
And I read the definition [here](https://scvelo.readthedocs.io/DynamicalModeling/#Latent-time)

Can you infer states between clusters just using it ? What conclusion can you have from this stuff only.
Can you say from that there is some gene expressions going up/down from one to other cluster, and that's all ?
And so what's the link with the spliced/unspliced ratio.

By the way, changing the root cell, can change a lot how the latent time is dispatched on the plot.

I do understand that we need to take a look at the phase portrait to understand the real thing behind the arrow.
We need to have characteristic almond/football shape to conclude something in fact otherwise it's bullshit. 
So latent time at the beginning is kind of counterintuitive , or useful for what ?

**Comment by GeorgeAlehandro** [comment]
![image](https://user-images.githubusercontent.com/74742818/233392992-d62cfdc6-cfb2-43c8-8f1f-1eccf48c06a9.png)
This is the error thrown because of pandas version at dyn.pp.recipe_monocle(adata)
